File: app/src/main/java/com/example/admin/myapplication/Model/RetrofitInstance.java
package com.example.admin.myapplication.Model;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitInstance {
    private static final String BASE_URL = "https://news-at.zhihu.com/api/3/news/";
    private static RetrofitInstance retrofitInstance;
    private Api api;

    private RetrofitInstance() {
        Retrofit retrofit = new Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .baseUrl(BASE_URL)
                .build();
        api = retrofit.create(Api.class);
    }

    public static RetrofitInstance getInstance() {
        if (retrofitInstance == null) {
            synchronized (RetrofitInstance.class) {
                if (retrofitInstance == null) {
                    retrofitInstance = new RetrofitInstance();
                }
            }
        }
        return retrofitInstance;
    }

    public Call<InfoBean> getLatestCall() {
        return api.getLatest();
    }

    public Call<InfoBean> getBeforeCall(String date) {
        return api.getBefore(date);
    }
}

File: app/src/main/java/com/example/admin/myapplication/Model/RetrofitInstance.kt
package com.example.admin.myapplication.Model

import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitInstance private constructor() {
    private val api: Api

    val latestCall: Call<InfoBean>
        get() = api.latest

    init {
        val retrofit = Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .baseUrl(BASE_URL)
                .build()
        api = retrofit.create(Api::class.java!!)
    }

    fun getBeforeCall(date: String): Call<InfoBean> {
        return api.getBefore(date)
    }

    companion object {
        private val BASE_URL = "https://news-at.zhihu.com/api/3/news/"
        private var retrofitInstance: RetrofitInstance? = null

        val instance: RetrofitInstance
            get() {
                if (retrofitInstance == null) {
                    synchronized(RetrofitInstance::class.java) {
                        if (retrofitInstance == null) {
                            retrofitInstance = RetrofitInstance()
                        }
                    }
                }
                return retrofitInstance
            }
    }
}
